apps/src/code-studio/levels/levelGroup.js

```diff
@@ -139,10 +138,8 @@ window.initLevelGroup = function (
       // Submit what we have, and when that's done, go to the next page of the
       // long assessment.  Cancel any pending throttled attempts at saving state.
       throttledSaveAnswers.cancel();
-      saveAnswers(function () {
-            changePage(targetPage);
-          },
-          submitSublevelResults);
+      const afterSave = () => changePage(targetPage);
+      submitSublevelResults(() => saveAnswers(afterSave));
```

Comment:
Here we used to call saveAnswers with a completion of submitSublevelResults and a beforeHook of submitSublevel results. The flow then was
- Call saveAnswers, which calls beforeHook/submitSublevelResults
- Upon finished it calls the main content of saveAnswers (sendResultsCompletion)
- sendResultsCompletion calls sendReport, passing it changePage as a completion

The new flow is:
- Call submitSubLevelResults
- Upon finished, it calls saveAnswers
- This calls sendReport, passing it changePage as a completion.